src/metabase/driver.clj

```diff
@@ -1270,11 +1270,12 @@
   dispatch-on-initialized-driver
   :hierarchy #'hierarchy)
 
-(defmulti rename-table!
-  "Rename a table from `old-table-name` to `new-table-name`. Table names may be qualified by schema e.g.
-
-    schema.table"
-  {:added "0.57.0", :arglists '([driver db-id old-table-name new-table-name])}
+(defmulti swap-table!
```

Comment:
I was thinking something simpler and less opinionated:

```clojure
(rename-tables! driver db-id {old->new})
```

We'd call this with `{temp-new target-name, target-name temp-old}`, and then call drop separately (it would never be in the same tx anyway).

Reply:
I guess something that sucks with a map is that some databases might require you to sequence the renames in a certain order. Maybe it's easier if we assume the caller has figured that out already, trivial for our case.

So maybe an association list is simpler `[[old new] [old new]]`.